pkgs/build-support/dotnet/fetch-nupkg/overrides.nix

```diff
@@ -17,6 +17,28 @@
   #     buildInputs = old.buildInputs or [ ] ++ [ hello ];
   #   });
 
+  "Avalonia" =
+    package:
+    package.overrideAttrs (
+      old:
+      let
+        compareVersion = builtins.compareVersions old.version;
+      in
+      # Versions between 11.1.0 and 11.2.0-beta1 (inclusive) need the fix from https://github.com/AvaloniaUI/Avalonia/pull/16835
+      lib.optionalAttrs
+        (compareVersion "11.1.0" >= 0 && (compareVersion "11.2.0" < 0 || old.version == "11.2.0-beta1"))
```

Comment:
This could become fragile, if the fix is backported to 11.1.x as we'd have:

- 11.1.{0-3} needs-patch
- 11.1.4 fixed (backport)
- 11.2.0-beta1 needs-patch
- 11.2.0 fixed

For now, this is a hypothetical though...

Since it's a short version range, maybe we could just implement a whitelist instead of doing semver comparisons?

```nix
elem old.version [
  "11.1.0"
  "11.1.1"
  "11.1.2"
  "11.1.3"
  "11.2.0-beta1"
]
```

Reply:
I only found one other affected packgae in nixpkgs ([retrospy](https://github.com/NixOS/nixpkgs/blob/9654ce30c522da850f51a56981bde734b7ce4aaf/pkgs/by-name/re/retrospy/deps.nix#L5)), which is using the exact same version as nexusmods-app.

I also found [beatsabermodmanager](https://github.com/NixOS/nixpkgs/blob/9654ce30c522da850f51a56981bde734b7ce4aaf/pkgs/by-name/be/beatsabermodmanager/deps.nix#L5) using a commit, not sure if it is affected or not.

If that's all we need to worry about, then maybe the version matching here doesn't need to be perfect. Let me know which approach you prefer; semver vs whitelist. And how robust each should be.

Reply:
I'm pretty sure I tested `beatsabermodmanager`, so it's probably fine.  side note: if you run `fetch-deps` on it, it fails to build afterward :shrug:  

Was `retrospy` also broken?  I thought I had tested everything.

I don't really have a strong opinion on the version thing.  In either case it'll only need to be touched when someone adds or updates a package and it's broken.

Reply:
Yeah, `retrospy` is definitely broken.  I mentioned it in the upgrade PR, but this PR should fix it.

Reply:
> Was retrospy also broken? I thought I had tested everything.

It's using an Avalonia version that has the issue, but maybe they're unafected by it because they're not using whatever XAML feature is broken?

I haven't tested, I was just greping the codebase to get a feel for if other packages are using affected Avalonia versions 🙂 

At a minimum, it seems we can just do `optionalAttrs (old.version == "11.1.3")`, if we only care about fixing the versions in use by nixpkgs packages.

If this is the kinda thing you'd remove when nexusmods-app bumps to an unaffected Avalonia version, perhaps we should keep it simple and take that approach?

Reply:
> At a minimum, it seems we can just do optionalAttrs (old.version == "11.1.3"), if we only care about fixing the versions in use by nixpkgs packages.

I think if we did the work to identify released versions that are broken, we might as well put them in at list.

> If this is the kinda thing you'd remove once nexusmods-app bumps to an unaffected Avalonia version, perhaps we should keep it simple and take that approach?

I don't think I'd be motivated to strip it out in a hurry.  It could still be useful outside of nixpkgs.

Reply:
Ok, thanks for your feedback! I've extracted the fix into https://github.com/NixOS/nixpkgs/pull/340493 so it can be reviewed independently of this PR.